components/camel-file-watch/src/main/java/org/apache/camel/component/file/watch/FileWatchConsumer.java

```diff
@@ -0,0 +1,241 @@
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.camel.component.file.watch;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Objects;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.concurrent.TimeUnit;
+
+import io.methvin.watcher.DirectoryChangeEvent;
+import io.methvin.watcher.DirectoryChangeListener;
+import io.methvin.watcher.DirectoryWatcher;
+import org.apache.camel.Exchange;
+import org.apache.camel.Message;
+import org.apache.camel.Processor;
+import org.apache.camel.RuntimeCamelException;
+import org.apache.camel.component.file.watch.constants.FileEvent;
+import org.apache.camel.component.file.watch.utils.PathUtils;
+import org.apache.camel.support.DefaultConsumer;
+import org.apache.camel.util.AntPathMatcher;
+
+/**
+ * The file-watch consumer.
+ */
+public class FileWatchConsumer extends DefaultConsumer {
+
+    private ExecutorService watchDirExecutorService;
+    private ExecutorService pollExecutorService;
+    private LinkedBlockingQueue<FileEvent> eventQueue;
+    private Path baseDirectory;
+    private AntPathMatcher antPathMatcher;
+    private DirectoryWatcher watcher;
+
+    public FileWatchConsumer(FileWatchEndpoint endpoint, Processor processor) {
+        super(endpoint, processor);
+        if (getEndpoint().getQueueSize() <= 0) {
+            eventQueue = new LinkedBlockingQueue<>();
+        } else {
+            eventQueue = new LinkedBlockingQueue<>(getEndpoint().getQueueSize());
+        }
+
+        antPathMatcher = new AntPathMatcher();
+        baseDirectory = Paths.get(getEndpoint().getPath());
+    }
+
+    @Override
+    protected void doStart() throws Exception {
+        super.doStart();
+
+        if (!Files.exists(baseDirectory)) {
+            if (getEndpoint().isAutoCreate()) {
+                baseDirectory = Files.createDirectories(baseDirectory);
+            } else {
+                throw new RuntimeCamelException("Path must exists when autoCreate = false");
+            }
+        }
+
+        if (!Files.isDirectory(baseDirectory)) {
+            throw new RuntimeCamelException(String.format("Parameter path must be directory, %s given", baseDirectory.toString()));
```

Comment:
We tend to use IllegalArgumentException for these kind

Reply:
Thanks, replaced.